src/Psalm/Internal/Type/Comparator/ArrayTypeComparator.php

```diff
@@ -123,10 +123,6 @@ public static function isContainedBy(
         }
 
         foreach ($input_type_part->type_params as $i => $input_param) {
-            if ($i > 1) {
```

Comment:
why did you remove that one?

Reply:
It causes the following error:
> TypeDoesNotContainType - src/Psalm/Internal/Type/Comparator/ArrayTypeComparator.php:126:17 - Type int<0, 1> for $i is never >1

Reply:
yep but that's wrong. This loop can go past one

Reply:
That must be one of the reasons I skipped tests and did not try analyzing assertions inside loops. Unfortunately, this is not ready

Reply:
What would cause the loop to go past one since the `type_params` property has a type of `array{Union, Union}`? And what is the `break` needed for that isn't covered by a test?

Reply:
Oh yeah, I guess you're right. I thought it was just a performance thing but the loop can't go over 1 indeed.

Array count has improved a lot recently, maybe it has fixed things that were broken when I decided to skip processing assertions in loop